--- a/cpp/ygo/field.cpp
+++ b/cpp/ygo/field.cpp
@@ -2383,7 +2383,7 @@
 	if(effect_replace_check(EFFECT_LPCOST_REPLACE, e))
 		return TRUE;
 	//cost[playerid].amount += val;
-	if(val <= player[playerid].lp)
+	if(val < player[playerid].lp)
 		return TRUE;
 	return FALSE;
 }
--- a/cpp/ygo/operations.cpp
+++ b/cpp/ygo/operations.cpp
@@ -719,7 +719,7 @@
 		e.reason_player = playerid;
 		core.select_options.clear();
 		core.select_effects.clear();
-		if(val <= player[playerid].lp) {
+		if(val < player[playerid].lp) {
 			core.select_options.push_back(11);
 			core.select_effects.push_back(nullptr);
 		}
